components/movementsensor/gpsnmea/component.go

```diff
@@ -17,81 +14,27 @@ import (
 	"go.viam.com/rdk/spatialmath"
 )
 
-// DataReader represents a way to get data from a GPS NMEA device. We can read data from it using
-// the channel in Messages, and we can close the device when we're done.
-type DataReader interface {
-	Messages() chan string
-	Close() error
-}
-
 // NMEAMovementSensor allows the use of any MovementSensor chip via a DataReader.
 type NMEAMovementSensor struct {
 	resource.Named
 	resource.AlwaysRebuild
-	cancelCtx               context.Context
-	cancelFunc              func()
-	logger                  logging.Logger
-	cachedData              rtkutils.CachedData
-	activeBackgroundWorkers sync.WaitGroup
-
-	dev DataReader
+	logger     logging.Logger
+	cachedData *rtkutils.CachedData
 }
 
 // NewNmeaMovementSensor creates a new movement sensor.
 func NewNmeaMovementSensor(
```

Comment:
I'd prefer the driver creator functions be unexported from within the drivers themselves, if that is possible.